### PR TITLE
[SYNPY-1237] Add mixed-integer-float data type

### LINKED ISSUE #940: as_table_columns function is mishandling mixed data types
# Bug Report

I noticed that we could not submit manifest as a table if the manifest has mixed integer and float data type. See the issue [here](https://github.com/Sage-Bionetworks/schematic/issues/870)

After digging into the code, I noticed that this is because `as_table_columns` function that we imported from `table.py` to schematic could not interpret mixed data types correctly. When a column contains an integer and a float, `infer_dtype` from pandas API would interpret the data type as `mixed-integer-float`. But since it doesn't fall into any of the data type defined in here: 

```
PANDAS_TABLE_TYPE = {
    'floating': 'DOUBLE',
    'decimal': 'DOUBLE',
    'integer': 'INTEGER',
    'boolean': 'BOOLEAN',
    'datetime64': 'DATE',
    'datetime': 'DATE',
    'date': 'DATE',
}
```
the code would still treat this data type as string. 

Based on the documentation that I found [here](https://python-docs.synapse.org/build/html/Table.html#synapseclient.table.Column), Synapse table columns have the following data type: “STRING”, “DOUBLE”, “INTEGER”, “BOOLEAN”, “DATE”, “FILEHANDLEID”, “ENTITYID”, “LINK”, “LARGETEXT”, “USERID” . Can we then have a new one called "NUMERIC" to handle columns with a mixed of "integer" and "float"? Or can we at least interpret the mixed of integer and float as "DOUBLE" so that it won't get interpreted as string?

### PR DESCRIPTION
**Pull request by thomasyu888**
Fixes #940 

@linglp can you test this by installing my branch locally with schematic?

### CODE REVIEW

**Comment by milen-sage** [comment]
@GiaJordan that's related to the conversation we had on Fri. Perhaps would avoid a workaround on our end if it works.

@thomasyu888 if this works, when would the feature get in a new release of the client?

**Comment by thomasyu888** [comment]
@milen-sage I'm trying to push this into v2.7, which I hope to release in the coming weeks.

**Comment by GiaJordan** [comment]
Thanks @thomasyu888!
@milen-sage I'll add this to my list and take a look

**Comment by GiaJordan** [comment]
> @thomasyu888 Confirming the fix works; was able to successfully submit valid_test_manifest.csv.
> Thank you!

File: tests/unit/synapseclient/unit_test_tables.py [hunk 1]
```diff
@@ -411,6 +411,15 @@ def test_pandas_to_table_convert__integer_data_type():
         assert col.columnType == "INTEGER"
 
 
+def test_pandas_to_table_convert__mixed_int_float_data_type():
+    test_df = pd.DataFrame({'int1': [1, 2, 990]})
+    schema = Schema(name="Foo", parent="syn12345")
+    with patch.object(pd.api.types, "infer_dtype", return_value="mixed-integer-float"):
```
[the remark this answers is not in the document]
Comment (GiaJordan) [reply tests/unit/synapseclient/unit_test_tables.py:419]: Do we also need to include a `float` entry along with the `ints` in the test df to model the dataframes the issue was first experienced on?

**Review by GiaJordan — APPROVED** [review]

**Pull request merged**